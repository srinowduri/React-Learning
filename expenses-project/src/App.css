.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #03115f;
  min-height: 8vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 40px;
  font-weight: bold;
  color: white;
  font-stretch: wider;
  font-family: Arial, Helvetica, sans-serif;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
