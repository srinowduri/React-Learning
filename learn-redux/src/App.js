// import './App.css';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {increment, decrement} from './actions';
function App() {
  const counter = useSelector(state => state.counter);
  const isLogged = useSelector(state => state.isLogged);
  const dispatch = useDispatch();
  return (
    <div className="App">
     <h1> Counter {counter} </h1>
     <button onClick={() => dispatch(increment(5))}>+</button>
     <button onClick={() => dispatch(decrement(5))}>-</button>
     {isLogged ? <h3>Information is not available</h3> : ''}

    </div>
  );
}

export default App;
