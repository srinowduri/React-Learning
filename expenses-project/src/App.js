import './App.css';
import ExpensesList from './Components/Expenses-list/Expenses-list';


function App() {
  return (
    <div className="App">
      <header className="App-header">Home Expenses</header>
      <ExpensesList></ExpensesList>
    </div>
  );
}

export default App;
