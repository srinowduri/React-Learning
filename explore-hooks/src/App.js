import './App.css';
import { Button } from './Button';
import { ButtonUseStatehook } from './Button_useSatehook';

function App() {
  return (
    <div className="App">
      <Button></Button>
      <ButtonUseStatehook></ButtonUseStatehook>
    </div>
  );
}

export default App;
