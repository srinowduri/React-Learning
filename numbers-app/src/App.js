import './App.css';
import NumbersData from './Data/numbersData';

function App() {
  return (
    <div className="App">
      <header className="App-header"></header><br></br>
      <NumbersData></NumbersData>
    </div>
  );
}

export default App;
