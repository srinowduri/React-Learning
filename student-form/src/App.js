import './App.css';
// import AddStudent from './components/Add-Student/AddStudent';
import Layout from './components/Layout/Layout';
import Logo from './components/Logo/Logo';
import { BrowserRouter } from 'react-router-dom';

function App() {
  return (
    <BrowserRouter>
    <div className="App">
      <header className="App-header"> <Logo /> SriSri Swaralaapana</header>
      <Layout>
          {/* <AddStudent></AddStudent> */}
      </Layout>
    </div>
    </BrowserRouter>
  );
}

export default App;
