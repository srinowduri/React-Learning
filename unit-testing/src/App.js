import React from 'react';
import './App.css';
import Practice from './Practice';

function App () {
    return (
      <div className="App">
        <header className="App-header">
          <h1>React Unit Testing</h1>
          <Practice></Practice>
        </header>
      </div>
    ); 
}

export default App;
