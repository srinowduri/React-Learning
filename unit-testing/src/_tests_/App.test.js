import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Add Student', () =>{
    it('should render Practice Component', () => {
        let container = shallow(<App />)
        //    console.log(container);
        expect(container.getElements()).toMatchSnapshot();
    })
})
