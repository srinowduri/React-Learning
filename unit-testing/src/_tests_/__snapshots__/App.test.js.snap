// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Student should render Practice Component 1`] = `
Array [
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1>
        React Unit Testing
      </h1>
      <Practice />
    </header>
  </div>,
]
`;
